packages/flutter/lib/src/rendering/sliver_grid.dart

```diff
@@ -59,11 +59,19 @@ class SliverGridGeometry {
   /// leading edge of the parent.
   double get trailingScrollOffset => scrollOffset + mainAxisExtent;
 
-  /// Returns a tight [BoxConstraints] that forces the child to have the
+  /// Returns [BoxConstraints] that will be tight if the [SliverGridLayout] has
+  /// provided fixed extents, forcing the child to have the
   /// required size.
+  ///
+  /// If the [mainAxisExtent] is [double.infinity] the child will be allowed to
+  /// chose its own size in the main axis. Similarly, an infinite
+  /// [crossAxisExtent] will result in the child sizing itself in the cross
+  /// axis. Otherwise, the provided cross axis size or the
+  /// [SliverConstraints.crossAxisExtent] will be used to create tight
+  /// constraints in the cross axis.
   BoxConstraints getBoxConstraints(SliverConstraints constraints) {
     return constraints.asBoxConstraints(
-      minExtent: mainAxisExtent,
+      minExtent: mainAxisExtent.isFinite ? mainAxisExtent : 0,
       maxExtent: mainAxisExtent,
       crossAxisExtent: crossAxisExtent,
     );
```

Comment:
This has been updated, you can copy paste the function into your branch. Probably simpler than merging if there are merge conflicts.